clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsLoadBalancerProvider.java

```diff
@@ -92,27 +106,116 @@ public String getCloudProvider() {
 
   @Override
   public Item get(String name) {
-    return null; // TODO - Implement this.
+    // unused method
+    throw new UnsupportedOperationException("Not implemented.");
```

Comment:
Not sure if we want to throw an exception here. Previously, this would just return null but allow users to continue, where as this would now throw an exception and potetially break and implementations that use this.

Reply:
My concern with always returning null is the client will proceed like they have correct information vs. understanding that this method should not be used. But I understand the concern about breaking folks; I'll take another look at where this (and the other such method) may cause problems and add more info here.

Reply:
Update: going back to returning `null` but adding more descriptive comments. [Other](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/provider/view/CloudFoundryLoadBalancerProvider.java#L61-L70) [providers](https://github.com/spinnaker/clouddriver/blob/29c5c1435ab0c750c4b491208f600647a3e98901/clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/view/DcosLoadBalancerProvider.groovy#L78-L89) [vary](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/provider/view/AppengineLoadBalancerProvider.groovy#L37-L51) between returning null and `OperationNotSupportedException`, indicating they are not required by deck, however hitting the [associated endpoints](https://github.com/spinnaker/gate/blob/master/gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/LoadBalancerController.groovy#L39-L56) directly now results in a 500. Would rather leave existing behavior intact if both have precedence.

Reply:
I'm not opposed to returning the `OperationNotSupportedException`, but we may want to open an issue, and let the community know prior to making the change.